src/lib/core/events.ts

```diff
@@ -0,0 +1,42 @@
+import events from "events";
+
+const COMMAND_PREFIX = "request";
+
+class EventEmitter extends events.EventEmitter {
+
+  public oncePromised(event: string | symbol, listener: (...args: any[]) => void = () => undefined) {
+    return new Promise<any[]>((resolve) => {
+      this.once(event, (...args: any[]) => {
+        listener(...args);
+        resolve(args);
+      });
+    });
+  }
+
+  public onPromised(event: string | symbol, listener: (...args: any[]) => void = () => undefined) {
+    return new Promise<any[]>((resolve) => {
+      this.on(event, (...args: any[]) => {
+        listener(...args);
+        resolve(args);
+      });
+    });
+  }
+
+  public request(event: string, ...args: any[]) {
+    return this.emit(`${COMMAND_PREFIX}:${event}`, ...args);
+  }
+
+  public setCommandHandler(event: string, listener: (...args: any[]) => void = () => undefined) {
+    const commandName = `${COMMAND_PREFIX}:${event}`;
+    this.removeAllListeners(commandName);
+    return this.onPromised(commandName, listener);
+  }
+
+  public setCommandHandlerOnce(event: string, listener: (...args: any[]) => void = () => undefined) {
+    return this.oncePromised(`${COMMAND_PREFIX}:${event}`, listener);
+  }
+}
+
+EventEmitter.prototype.setMaxListeners(350);
+
+module.exports = EventEmitter;
```

Comment:
Instead of going with commonjs syntax here, I'd vote for exporting symbols directly e.g.

```
export class EventEmitter extends Foo {

}
```

So we can then do named imports a la 

```
import { EventEmitter } from 'lib/core/events';
```

Reply:
I just wanted to avoid to modify all classes using events but will do